### PR TITLE
Asyncify eth.get_transaction_receipt and eth.wait_for_transaction_receipt

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Added async `eth.get_transaction_receipt` and `eth.wait_for_transaction_receipt`

Related to Issue #1413 
Closes Issue #2129

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: web3/eth.py [hunk 1]
```diff
@@ -277,6 +277,17 @@ def call_munger(
         mungers=None,
     )
 
+    _get_transaction_receipt: Method[Callable[[_Hash32], TxReceipt]] = Method(
+        RPC.eth_getTransactionReceipt,
+        mungers=[default_root_munger]
+    )
+
+    # _get_transaction_receipt_awaitable: Method[Callable[[_Hash32],
```
Comment (kclowes) [inline web3/eth.py:285]: this can go :)

**Review by kclowes — APPROVED** [review]
This looks good to me! Looks like there is a left over comment that can be removed. Did you manually test it to make sure it works like you think it should?

File: web3/eth.py [hunk 2]
```diff
@@ -403,6 +414,36 @@ async def get_transaction_count(
...
+                while True:
+                    try:
+                        tx_receipt = await self._get_transaction_receipt(  # type: ignore
+                            transaction_hash
+                        )
+                    except TransactionNotFound:
+                        tx_receipt = None
+                    if tx_receipt is not None:
+                        break
+                    _timeout.sleep(poll_latency)
```
Comment (kclowes) [inline web3/eth.py:436]: Sorry to pop up again after approving this, but I think we want to be able to await this so it's not blocking. I think the most straightforward route is to get rid of the timeout context and use `await asyncio.sleep` here. If you want to see if it makes a difference at all, you could add it to our benchmarking suite and see what happens :)

File: web3/eth.py (hunk 1 above)
Comment (pacrob) [reply web3/eth.py:285]: Yup, thanks for catching

File: web3/eth.py (hunk 2 above)
Comment (pacrob) [reply web3/eth.py:436]: Makes sense. _timeout.sleep replaced with await asyncio.sleep

**Pull request merged**